### PR TITLE
Add reset password button to ops project list

### LINKED ISSUE #4844: Check or reset the ID/PW of a project once registered when forgetting them
**What would you like to be added**:

It would be nice to be able to check or reset ID/PW in case we forgot them.

**Why is this needed**:
We can't log in when we forget them.

### PR DESCRIPTION
**Pull request by dgannon991**
**What this PR does / why we need it**:
Adds a reset password button to the project list in the ops view

**Which issue(s) this PR fixes**:

Fixes #4844

**Does this PR introduce a user-facing change?**:
Yes, a new button and column in the ops list

- **How are users affected by this change**: the OPs UI is updated
- **Is this breaking change**: No
- **How to migrate (if breaking change)**: N/A

Just pushing up a draft for this at the moment to get some early feedback. I'll have a look through to see how/if we're testing the ops side of things later, but wanted to provide an early view of the UI. 

image

### CODE REVIEW

**Comment by dgannon991** [comment]
@khanhtc1202 & @ffjlabo - forgot that it wouldn't notify you in draft mode :D Nothing urgent, just hoped to get some feedback on the UI before writing up the tests. Cheers!

**Comment by ffjlabo** [comment]
@dgannon991 
Thank you so much for the quick response!! The button UI is LGTM for me :)

Thanks to your PR, I noticed this action is extremely dangerous because the control plane operator might reset for other projects accidentally. 👀 
It would be nice to prevent users from accidentally initializing static admins for other projects.

So, how about adding a confirmation screen?
e.g. the project list page -> reset password confirmation page -> success page

Reset password confirmation page like this↓ (the sentences are the draft, and I want more correct them 🙏 )
```
Reset Static Admin for the project XXX.
You must check the project name before you click the button.

"Reset button"
```

**Comment by khanhtc1202** [comment]
@dgannon991 @ffjlabo 
How about asking the user to reinstate the project name as part of the reset password confirmation page? (Cloud service like AWS use that method as re-check step if I remember correctly)

**Comment by dgannon991** [comment]
I like the sound of that. I've got some down time over the next couple of days, so I'll give it a go and let you know :)

**Comment by ffjlabo** [comment]
@dgannon991 @khanhtc1202 
> How about asking the user to reinstate the project name as part of the reset password confirmation page? (Cloud service like AWS use that method as re-check step if I remember correctly)

Thank you both 🙏 I agree with it!

**Comment by dgannon991** [comment]
Hi Guys, 
I've added a confirmation page, and you have to retype the project ID. 
One note though, the user experience is not great when you get things wrong (due to the server just returning error responses instead of markup.) Are we happy with this, as the ops pages are for technical users?
Cheers!

**Comment by ffjlabo** [comment]
@dgannon991 
Thank you for the nice commitment :)
I also agree the ops pages are plain because these are for technical users.

[IMO] If possible, it would be nice to show the Reset page with the failed message, or the link to the Reset page.

**Project Page**
Cursor_と_localhost_9082_projects

**Fail**
Cursor_と_localhost_9082_projects_reset-password_ID_test

**Succsess**
localhost_9082_projects_reset-password_ID_test

**Comment by dgannon991** [comment]
Absolutely. I'll give that a whirl over the weekend. Cheers for the continued feedback on this one guys :)

**Comment by dgannon991** [comment]
Morning! I've updated this to show a nice error message on the confirmation page. Do let me know if it's too big and red! I looked at the material UI library we use in the normal frontend for inspiration, but it is the first think like it on the ops pages!

**Comment by ffjlabo** [comment]
Hi @dgannon991 
Sorry for the late reply. Currently doing the other task.
So plz wait a moment 🙏

**Comment by dgannon991** [comment]
Hi @ffjlabo,

Not a problem at all, the other task looks mega! No rush from my side at all :) 

D

**Comment by dgannon991** [comment]
Hi Guys,
Just working on the coverage now. I've brought it up, but I think it will now be easy to really get it much higher, so I'll have another go tomorrow night at getting it there :) 
Cheers!
David

**Comment by dgannon991** [comment]
Hi Guys,
Managed to get a few more tests written, and I've got pretty much everything I added covered. It does use gomock (which is a shame) but I couldn't think of a better way to test without actually creating a project store!
Let me know what you think!
D

File: pkg/app/ops/handler/handler_test.go [hunk 1]
```diff
@@ -16,78 +16,213 @@ package handler
...
+	"strings"
 	"testing"
 
-	"github.com/stretchr/testify/assert"
-
+	"github.com/golang/mock/gomock"
+	"github.com/pipe-cd/pipecd/pkg/config"
 	"github.com/pipe-cd/pipecd/pkg/model"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
```
Comment (khanhtc1202) [inline pkg/app/ops/handler/handler_test.go:32]:
```suggestion
	"github.com/golang/mock/gomock"
	"github.com/stretchr/testify/assert"
	"go.uber.org/zap"
	
	"github.com/pipe-cd/pipecd/pkg/config"
	"github.com/pipe-cd/pipecd/pkg/model"
```
nits

File: pkg/app/ops/handler/handler.go [hunk 2]
```diff
@@ -152,6 +157,115 @@ func (h *Handler) handleListProjects(w http.ResponseWriter, r *http.Request) {
...
+}
+
+func getStaticUsername(project *model.Project) string {
+	var username = ""
+	staticAdmin := project.GetStaticAdmin()
+	if staticAdmin != nil {
+		username = staticAdmin.GetUsername()
+	}
+	return username
+}
```
Comment (khanhtc1202) [inline pkg/app/ops/handler/handler.go:205]: nits, this should be a receiver function and under the model package. Something like
```golang
func (p *Project) GetStaticUsername() { ... }
```

**Review by khanhtc1202 — DISMISSED** [review]
I checked the behavior, and it works as we expected. Thanks a lot, @dgannon991 👍 
I left some nits, but they can mostly be fixed later by separated PRs.

LGTM 👍

**Comment by dgannon991** [comment]
Hi Guys, more than happy to address these here if you'd like? LMK.

**Review by Warashi — APPROVED** [review]
I also checked the behavior, and it worked as expected :+1:
I have no comments other than comments by @khanhtc1202 
LGTM :+1:

**Comment by khanhtc1202** [comment]
> Hi Guys, more than happy to address these here if you'd like? LMK.

@dgannon991 For sure, let's do as you want 👍

**Comment by Warashi** [comment]
Thank you for saying that.
I dismissed my approval for this PR not to make this PR merge by mistake, but it does not mean the code is not good.

**Comment by dgannon991** [comment]
Nice one! I’m AFK until Thursday night, but I’ll get it done Friday morning! Cheers all.

**Comment by dgannon991** [comment]
Hi Guys,
The git history got a bit squirrely there, apologies. I rebased on my local master, but didn't realise it was stale! I'd definitely suggest a squash commit on this one! 
I've made the changes, and run the tests, and things still look good. I'll stand everything up when I get a second tomorrow AM and give a final update :)
Cheers all.

**Comment by dgannon991** [comment]
Hi Guys,
This looks better now, removed all the silly extra commits! I've given it a run through and it all works as expected :)
Cheers!
David

File: pkg/model/project.pb.go [hunk 3]
```diff
@@ -305,6 +305,15 @@ func (x *Project) GetStaticAdminDisabled() bool {
 	return false
 }
 
+func (x *Project) GetStaticAdminUsername() string {
```
Comment (Warashi) [inline pkg/model/project.pb.go:308]: This file is generated by protoc-gen-go, so please move this method to `project.go` in the same directory.

**Comment by Warashi** [comment]
I checked the behavior, and it worked fine.
I commented nits. Please re-request a review or mention us in the comments when you resolve that.

**Review by khanhtc1202 — APPROVED** [review]
Thank you 💯

**Review by ffjlabo — APPROVED** [review]
Thank you!

**Pull request merged**